### PR TITLE
Documentation/admin/*-onboard.md: RBAC/LDAP Fix broken links, reflect UI changes

### PR DESCRIPTION
**Pull request by radhikapc**
[Under Construction]
Fix broken links, reflect UI changes (YAML editor for creating roles and role binding)

### CODE REVIEW

**Comment by robszumski** [comment]
Can you update the description of this? It seems like it adds a lot more than just broken links. Maybe break those out into separate commits next time?

**Comment by radhikapc** [comment]
@robszumski i did a commit --amend. i will change the description

**Comment by zbwright** [comment]
just on first glance - it doesn't look like any of the code snippets are formatted in these files.

**Comment by radhikapc** [comment]
[do not merge] - am still working on it.

**Comment by radhikapc** [comment]
cc @kyoto

File: Documentation/admin/onboard-admin.md [hunk 1]
```diff
@@ -1,37 +1,142 @@
...
-7. Specify a name to identify subject.
-8. Click *Create Binding*.
+2. Select *Roles* under *Administration*.
+3. Select *all* from the drop-down given at the top of the page.
+4. Click *Create Role*.
+   A YAML editor screen is displayed.
+5. Make necessary edits.
+
+    The following shows an example of granting a user admin access to the cluster. The `sample-cluster-admin` role can run commands get, watch, and list the cluster. `*` stands for full access.
+
```
Comment (zbwright) [inline Documentation/admin/onboard-admin.md]: outdent the yaml content, please. a good example for formatting lives here: https://coreos.com/tectonic/docs/latest/tutorials/scale-app.html

File: Documentation/admin/onboard-admin.md [hunk 2]
```diff
@@ -1,37 +1,142 @@
...
+### Creating a Cluster Role Binding
+
+A Cluster Role Binding grants permissions to users in all namespaces across the entire cluster. Bind the `sample-cluster-admin` role appropriately to the privileges associated with a cluster role binding, `sample-admin`. `namespace` has been omitted from the configuration because Cluster Roles are not namespaced.
+
+1. Navigate to *Role Bindings* under *Administration*.
+2. Select a desired namespace from the drop-down given at the top of the page.
+3. Click *Create Binding*.
+   A YAML editor screen is displayed.
+
```
Comment (zbwright) [inline Documentation/admin/onboard-admin.md]:
```yaml (and outdent, as described earlier, please.)

File: Documentation/admin/onboard-admin.md (hunk 1 above)
Comment (radhikapc) [inline Documentation/admin/onboard-admin.md]: I know about triple quotes -- it doesn't work with listed items. I am investigating on fixing it.

File: Documentation/admin/onboard-admin.md [hunk 3]
```diff
@@ -1,37 +1,142 @@
...
+### Creating a Role
+
+For roles, use one of the default cluster roles, or create a new role by using either of the following in the *Roles* option under *Administration*:
+
+* YAML editor
+* Selecting a default role and changing the rules using the *Add Rules* option
+
+An example YAML file of namespace admin:
+
```
Comment (zbwright) [inline Documentation/admin/onboard-admin.md]: three ticks (```) and yaml (and outdent)

File: Documentation/admin/onboard-admin.md [hunk 4]
```diff
@@ -1,37 +1,142 @@
...
+1. Navigate to *Role Bindings* under *Administration*.
+2. Select a desired namespace from the drop-down given at the top of the page.
+3. Click *Create Binding*.
+
+   A YAML editor screen is displayed.
+4. Make necessary edits.
+
+    An example YAML file for a role binding for `kube-public` cluster:
+
```
Comment (zbwright) [inline Documentation/admin/onboard-admin.md]: three ticks (```) and yaml (and outdent)

File: Documentation/admin/onboard-service-account.md [hunk 5]
```diff
@@ -2,24 +2,22 @@
...
-A service account, `default`, by default is created by Tectonic. However, creating an additional service account is permitted.
 
 ## Creating an additional service accounts
 
 To create an additional service account, for example, an ingress role, either create a yaml file as follows or use the Tectonic console to create one. Given is an example service account for ingress.
 
-### Using an YAML file
+### Through command line
 
 1. Define the role, `ingress.yaml`, which gives administrative privileges to the service account within the default namespace:
```
Comment (zbwright) [inline Documentation/admin/onboard-service-account.md]: three ticks (```) and yaml (and outdent)

File: Documentation/admin/onboard-admin.md (hunk 3 above)
Comment (radhikapc) [inline Documentation/admin/onboard-admin.md]: plz see my previous comment.

File: Documentation/admin/onboard-service-account.md [hunk 6]
```diff
@@ -54,33 +52,115 @@ If multiple pods running in the same namespace require different levels of acces
...
-1. Log in to the Tectonic UI
-2. Navigate to *Role Bindings* under *Administration*.
-3. Click *Namespace Role Binding*.
-4. Select a desired namespace from the drop-down.
-5. Select a Role Name.
-   See [Default Roles in Tectonic][identity-management].
-6. Select *Service Account* from subject kind.
-7. Specify a a name to identify subject.
-8. Click *Create Binding*.
+   A YAML file for an example service account is given below. Note that  cluster roles are not namespaced.
```
Comment (zbwright) [inline Documentation/admin/onboard-service-account.md]: three ticks (```) and yaml (and outdent)

File: Documentation/admin/onboard-service-account.md [hunk 7]
```diff
@@ -54,33 +52,115 @@ If multiple pods running in the same namespace require different levels of acces
...
+
+1. Log in to the Tectonic UI.
+2. Navigate to *Role Bindings* under *Administration*.
+3. Select a desired namespace from the drop-down.
+4. Click *Create Binding*.
+   A YAML editor is displayed.   
+5. Make necessary edits.
 
-### Cluster-wide
+   Given below is a YAML file for a service account with permission to run get, watch, and list commands on the cluster.
```
Comment (zbwright) [inline Documentation/admin/onboard-service-account.md]: three ticks (```) and yaml (and outdent)

File: Documentation/admin/onboard-service-account.md [hunk 8]
```diff
@@ -54,33 +52,115 @@ If multiple pods running in the same namespace require different levels of acces
...
+
+1. Log in to the Tectonic UI.
+2. Navigate to *Roles* under *Administration*.
+3. Select a desired namespace from the drop-down.
+4. Click *Create Roles*.
+   A YAML editor is displayed.   
+5. Make necessary edits.
+
+   A YAML file for an example service account is given below:
+
```
Comment (zbwright) [inline Documentation/admin/onboard-service-account.md]: three ticks (```) and yaml (and outdent)

File: Documentation/admin/onboard-service-account.md [hunk 9]
```diff
@@ -54,33 +52,115 @@ If multiple pods running in the same namespace require different levels of acces
...
-6. Select *Service Acccount* from subject kind.
-7. Specify a a name to identify subject.
-8. Click *Create Binding*.
+3. Select a desired namespace from the drop-down.
+4. Click *Create Binding*.
+   A YAML editor is displayed.   
+5. Make necessary edits.
+
+   A YAML file for an example service account is given below:
+
```
Comment (zbwright) [inline Documentation/admin/onboard-service-account.md]: three ticks (```) and yaml (and outdent)

File: Documentation/admin/onboard-team.md [hunk 10]
```diff
@@ -4,45 +4,77 @@ A group can be assigned to a cluster or a specific namespace within a cluster. U
...
+2. Navigate to *Roles* under *Administration*.
+3. Select a desired namespace from the drop-down.
+4. Create a role by editing the YAML editor.   
+5. Navigate to *Roles* under *Administration*.
+6. Select a desired namespace from the drop-down.
+7. Click *Create Binding*.
+   A YAML editor is displayed.
+8. Make necessary edits to create the role binding.
+   An example YAML file that grant access to kube-system namespace:
+
```
Comment (zbwright) [inline Documentation/admin/onboard-team.md]: three ticks (```) and yaml (and outdent)

File: Documentation/admin/onboard-team.md [hunk 11]
```diff
@@ -4,45 +4,77 @@ A group can be assigned to a cluster or a specific namespace within a cluster. U
...
+2. Navigate to *Roles* under *Administration*.
+3. Select a desired namespace from the drop-down.
+4. Create a role by editing the YAML editor.   
+5. Navigate to *Roles* under *Administration*.
+6. Select a desired namespace from the drop-down.
+7. Click *Create Binding*.
+   A YAML editor is displayed.
+8. Make necessary edits to create the role binding.
+   An example YAML file that grant access to kube-system namespace. Note that "namespace" omitted because Cluster Roles are not namespaced.
+
```
Comment (zbwright) [inline Documentation/admin/onboard-team.md]: three ticks (```) and yaml (and outdent)

File: Documentation/admin/onboard-user.md [hunk 12]
```diff
@@ -1,34 +1,176 @@
...
+
+1. Log in to the Tectonic UI.
+2. Select *Roles* under *Administration*.
+3. Select *all* from the drop-down given at the top of the page.
+4. Click *Create Role*.
+   A YAML editor screen is displayed.
+5. Make necessary edits.
+
+  The following shows an example of granting a user basic access to the cluster. The `user-readonly` role can run commands get, watch, list, proxy, and redirect namespaces and pods in the given cluster.
+
```
Comment (zbwright) [inline Documentation/admin/onboard-user.md]: outdent, please. see the tutorials for an example.

File: Documentation/admin/onboard-user.md [hunk 13]
```diff
@@ -1,34 +1,176 @@
...
+Note that "namespace" is omitted because Cluster Roles are not namespaced.
+
+1. Navigate to *Role Bindings* under *Administration*.
+2. Select *all* from the drop-down given at the top of the page.
+3. Click *Create Binding*.
+   A YAML editor screen is displayed.
+4. Make necessary edits.
+
+    ```yaml
```
Comment (zbwright) [inline Documentation/admin/onboard-user.md]: outdent, please.

File: Documentation/admin/onboard-user.md [hunk 14]
```diff
@@ -1,34 +1,176 @@
...
+              apiGroup: rbac.authorization.k8s.io
+              ```
+
+5. Click *Create*.
+
+A Role Binding can also be associated to a Cluster Role to grant permissions to resources within a namespace as defined in the Role Binding. This allows administrators to define a set of common roles for the entire cluster, then reuse them within multiple namespaces.
+
+For instance, even though the following Role Binding refers to a Cluster Role, user will only be having read access to the `kube-system` namespace:
+
+```yaml
```
Comment (zbwright) [inline Documentation/admin/onboard-user.md]: outdent, please.

File: Documentation/admin/onboard-user.md [hunk 15]
```diff
@@ -1,34 +1,176 @@
...
+1. Log in to the Tectonic UI.
+2. Select *Roles* under *Administration*.
+3. Select *all* from the drop-down given at the top of the page.
+4. Click *Create Role*.
+   A YAML editor screen is displayed.
+5. Make necessary edits.
+
+  The following shows an example of granting a user basic access to the `kube-public` namespace. The `namespace-user` role can run commands get, watch, and list pods within `kube-public`.
+
+```yaml
```
Comment (zbwright) [inline Documentation/admin/onboard-user.md]: outdent, please.

File: Documentation/admin/onboard-user.md [hunk 16]
```diff
@@ -1,34 +1,176 @@
...
-8. Click *Create Binding*.
+3. Select a desired namespace from the drop-down given at the top of the page.
+4. Click *Create Binding*.
+
+   A YAML editor screen is displayed.
+5. Make necessary edits.
+
+  An example YAML file for a namespace role binding:
+
+```yaml
```
Comment (zbwright) [inline Documentation/admin/onboard-user.md]: outdent, please.

File: Documentation/admin/onboard-team.md [hunk 17]
```diff
@@ -4,45 +4,77 @@ A group can be assigned to a cluster or a specific namespace within a cluster. U
...
+        metadata:
+          name: example-role-binding
+        subjects:
+          - kind: Group
+            name: tstgroup
+            apiGroup: rbac.authorization.k8s.io
+        roleRef:
+          kind: ClusterRole
+          name: example-cluster-role
+          apiGroup: rbac.authorization.k8s.io
```
Comment (zbwright) [inline Documentation/admin/onboard-team.md]: three ticket (```)

File: Documentation/admin/onboard-team.md [hunk 18]
```diff
@@ -4,45 +4,77 @@ A group can be assigned to a cluster or a specific namespace within a cluster. U
...
+          name: example-role-binding
+          namespace: kube-system
+        subjects:
+          - kind: Group
+            name: tstgroup
+            apiGroup: rbac.authorization.k8s.io
+        roleRef:
+          kind: Role
+          name: example-role
+          apiGroup: rbac.authorization.k8s.io
```
Comment (zbwright) [inline Documentation/admin/onboard-team.md]: three ticks (```)

File: Documentation/admin/onboard-service-account.md [hunk 19]
```diff
@@ -54,33 +52,115 @@ If multiple pods running in the same namespace require different levels of acces
...
+        rules:
+          - verbs:
+              - get
+              - watch
+              - list
+            apiGroups:
+              - extensions
+            resources:
+              - ingress
+
```
Comment (zbwright) [inline Documentation/admin/onboard-service-account.md]: three ticks (```)

File: Documentation/admin/onboard-service-account.md [hunk 20]
```diff
@@ -54,33 +52,115 @@ If multiple pods running in the same namespace require different levels of acces
...
+      rules:
+        - verbs:
+            - get
+            - watch
+            - list
+          apiGroups:
+            - extensions
+          resources:
+            - ingress
+
```
Comment (zbwright) [inline Documentation/admin/onboard-service-account.md]: three ticks (```)

File: Documentation/admin/onboard-service-account.md [hunk 21]
```diff
@@ -54,33 +52,115 @@ If multiple pods running in the same namespace require different levels of acces
...
+        rules:
+          - verbs:
+              - get
+              - watch
+              - list
+            apiGroups:
+              - extensions
+            resources:
+              - ingress
+
```
Comment (zbwright) [inline Documentation/admin/onboard-service-account.md]: three ticks (```)

File: Documentation/admin/onboard-service-account.md [hunk 22]
```diff
@@ -54,33 +52,115 @@ If multiple pods running in the same namespace require different levels of acces
             image: nginx:1.7.9
           serviceAccountName: ingress # Specify the custom service account
```
Comment (zbwright) [inline Documentation/admin/onboard-service-account.md]: three ticks (```)

File: Documentation/admin/onboard-admin.md [hunk 23]
```diff
@@ -1,37 +1,142 @@
...
+         namespace: kube-public
+       subjects:
+         - kind: User
+           name: my-sample-group
+           apiGroup: rbac.authorization.k8s.io
+       roleRef:
+         kind: Role
+         name: sample-namespace-admin
+         apiGroup: rbac.authorization.k8s.io
+
```
Comment (zbwright) [inline Documentation/admin/onboard-admin.md]: three ticks (```)

File: Documentation/admin/onboard-admin.md [hunk 24]
```diff
@@ -1,37 +1,142 @@
...
+      name: sample-namespace-admin
+      namespace: kube-public      
+    rules:
+      - verbs:
+          - ''
+        apiGroups:
+          - ''
+        resources:
+          - ''
+
```
Comment (zbwright) [inline Documentation/admin/onboard-admin.md]: three ticks (```)

File: Documentation/admin/onboard-admin.md [hunk 25]
```diff
@@ -1,37 +1,142 @@
...
+         name: sample-role-binding-admin
+       subjects:
+         - kind: User
+           name: sample-admin
+           apiGroup: rbac.authorization.k8s.io
+       roleRef:
+         kind: ClusterRole
+         name: sample-cluster-admin
+         apiGroup: rbac.authorization.k8s.io
+
```
Comment (zbwright) [inline Documentation/admin/onboard-admin.md]: three ticks (```)

**Comment by sym3tri** [comment]
This pull request is too large to review as a whole. Can you please break this up into smaller more concise PRs, one for each type of change?

**Comment by radhikapc** [comment]
@sym3tri this isn't complete as Andy has made changes to what I have documented in this PR. I do not have a tectonic setup to test latest changes and make any changes! I also requested in the channel to clear EIP so that I can have a working setup. I pinged you and Andy too! Unless I have a working cluster there is no point going ahead with this PR because UI has changed as per UX team. Once I have a working cluster I can rework on it and submit smaller chunks.

**Comment by radhikapc** [comment]
closing as per @sym3tri direction. Will be reworking on it when i have a working cluster and creating smaller chunks of doc PR.

**Pull request closed without merging**